cts/cts-cli.in

```diff
@@ -386,7 +386,7 @@ function test_tools() {
     test_assert $CRM_EX_OK
 
     desc="Try to move a resource to its existing location"
-    cmd="crm_resource -r dummy --move --host node1"
+    cmd="crm_resource -r dummy --move --host-uname node1"
```

Comment:
Ouch ... I guess that means that despite pcmk__cmdline_preproc(), we still didn't keep perfect backward compatibility. :-( Too late to stop now, at least we never documented that "feature"

Reply:
Is there a reason we can't define `--host` as a hidden option (for compatibility) and define `--host-uname` as a regular argument, both mapped to the same variable?

I ran a quick test with the following entries and I didn't have a problem mapping two options to the same destination variable:
```
  { "host", 0, G_OPTION_FLAG_HIDDEN, G_OPTION_ARG_STRING, &host_uname, 0, NULL },
  { "host-uname", 0, 0, G_OPTION_ARG_STRING, &host_uname, "Host uname", NULL },
```

https://developer.gnome.org/glib/stable/glib-Commandline-option-parser.html#GOptionFlags

Reply:
It's not practical to add options for --ho, --hos, --host-, etc., so unless we find some known external code using an abbreviation, I'd rather just drop it

Reply:
There's nothing in pcmk__cmdline_preproc that handles this case.  I was going to say that I could probably add something to handle this, but it doesn't look like glib provides me any way to do it.  Once you put the GOptionEntry items in, there's not really any way to get them back out.  So, I can't iterate over them and see which one matches an abbreviated command line option.

Reply:
Right, my reference to pcmk__cmdline_preproc() was just intended to say we already bent over backward to make GOption compatible with getopt_long(), but it's just too different to be 100%

I prefer to preserve backward compatibility whenever possible but I'm less concerned about undocumented behaviors, so I think we can pretend the former behavior was a bug that we are now correcting